### PR TITLE
Set crossplane user agent on package requests

### PR DESCRIPTION
**Pull request by hasheddan**
### Description of your changes

Sets `crossplane/<version>` user agent header on all requests made for packages. This is more explicit than the `go-containerregistry/<version>` that is utilized currently. 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Ran against local registry and verified that user agent is set to `crossplane/v1.11.0-rc.0.27.gf447ebe1.dirty go-containerregistry/v0.9.0`.

**Update:** `go-containerregistry` currently does not allow removal of the default `go-containerregistry/<version>` `User-Agent` header, so the Crossplane one is appended for the time-being.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by nullable-eth — APPROVED** [review]
Looks good and is a useful distinction.

File: internal/transport/transport.go [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+func NewUserAgent(rt http.RoundTripper, userAgent string) *UserAgent {
+	return &UserAgent{
+		rt:        rt,
+		userAgent: userAgent,
+	}
+}
+
+// RoundTrip injects a User-Agent header into every request.
+func (u *UserAgent) RoundTrip(req *http.Request) (*http.Response, error) {
+	req.Header.Set("User-Agent", u.userAgent)
```
Comment (negz) [inline internal/transport/transport.go:46]: I'm guessing this will always be applied after and thus override the default user agent, right?

**Review by negz — APPROVED** [review]

File: internal/transport/transport.go (hunk 1 above)
Comment (hasheddan) [reply internal/transport/transport.go:46]: @negz right 👍🏻

File: internal/transport/transport.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/crossplane/crossplane/internal/version"
+)
+
+// DefaultUserAgent is the default User-Agent header that is set when making
+// HTTP requests for packages.
+var DefaultUserAgent = fmt.Sprintf("%s/%s", "crossplane", version.New().GetVersionString())
```
Comment (ytsarev) [inline internal/transport/transport.go:28]: Just to double-check, it should also expose UXP in the form of `up` in the version like `1.9.1-up.2` is that correct?

**Review by ytsarev — APPROVED** [review]

File: internal/transport/transport.go (hunk 2 above)
Comment (hasheddan) [reply internal/transport/transport.go:28]: @ytsarev that will be a downstream concern 👍🏻

**Pull request merged**